### DISCUSSION TITLE
Where does PATH come from OSX?

### DISCUSSION THREAD

**Discussion by newhook** [wezterm/wezterm#3905] [Q&A] [answered]
I was trying a launch menu to start a program, and it isn't in my PATH.

Looking at the wezterm log I see:

```
"PATH", value: "/usr/bin:/bin:/usr/sbin:/sbin"
```

Where is this PATH coming from? My ~/.zshrc has many more items. Do I need to add the PATH entries elsewhere?

**Comment by wez** [comment, marked as answer]
Finder has a minimal path. Finder is usually what launches wezterm and passes on its PATH.
Neither Finder nor wezterm use your shell when launching programs, so none of your shell configuration to change the path is visible to wezterm itself.  Normally that isn't an issue because wezterm launches your shell into a terminal, but if you need wezterm to launch something that is not in your path, you can use a fully qualified path in eg: the `default_prog` config, or you can adjust environment variables in wezterm.  See https://wezfurlong.org/wezterm/config/launch.html for more information on launching programs.

**Reply by XA21X** [reply]
Thanks. Example here worked for me: https://github.com/wez/wezterm/issues/3950#issuecomment-1753646215
